internal/handlers/common/filter_iterator.go

```diff
@@ -68,6 +89,138 @@ func (iter *filterIterator) Close() {
 	iter.iter.Close()
 }
 
+// evaluateExpr evaluates `$expr` operator and returns boolean indicating filter match.
+func evaluateExpr(doc, filter *types.Document) (bool, error) {
+	op, err := operators.NewExpr(filter)
+	if err != nil {
+		return false, processExprError(err)
+	}
+
+	v, err := op.Process(doc)
+	if err != nil {
+		return false, lazyerrors.Error(err)
+	}
+
+	switch v := v.(type) {
```

Comment:
This is a part I don't understand. What does it mean "evaluates"? It's not the calculation of expression result, right? As far as I understand the logic, this function returns true if the expression is valid. So, is it validation then? Should the function be called `validateExpr`?